app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/HorizontalSnapScrollView.kt

```diff
@@ -0,0 +1,267 @@
+            activeColumnIndex = col
+        )
+    }
+
+    private inner class OnTouchListener : View.OnTouchListener {
+        override fun onTouch(v: View, event: MotionEvent): Boolean {
+            logging.d(LOG_TAG, "onTouch")
+            if (gestureDetector.onTouchEvent(event)) {
+//                logging.d(LOG_TAG, "gesture detector consumed event")
+                return false
+            } else if (
+                event.action == MotionEvent.ACTION_UP ||
+                event.action == MotionEvent.ACTION_CANCEL
+            ) {
+                val scrollX = event.x.toInt()
+                val columnIndex = horizontalSnapScrollState.calculateOnTouchColumnIndex(scrollX)
+                scrollToColumn(columnIndex, false)
+                return true
+            }
+            return false
+        }
+    }
+
+    override fun onSizeChanged(width: Int, height: Int, oldWidth: Int, oldHeight: Int) {
+        logging.d(
+            LOG_TAG,
+            "onSizeChanged $oldWidth, $oldHeight, $width, $height getMW:$measuredWidth"
+        )
+        super.onSizeChanged(width, height, oldWidth, oldHeight)
+        val displayColumnCount = if (horizontalSnapScrollState.isRoomsCountInitialized()) {
+            calculateDisplayColumnCount(
+                measuredWidth,
+                horizontalSnapScrollState.roomsCount,
+                resources.getInteger(R.integer.max_cols),
+                resources.displayMetrics.density,
+                resources.getInteger(R.integer.min_width_dip)
+            )
+        } else {
+            1
+        }
+        horizontalSnapScrollState = horizontalSnapScrollState.copy(
+            logging = horizontalSnapScrollState.logging,
+            xStart = horizontalSnapScrollState.xStart,
+            displayColumnCount = displayColumnCount,
+            roomsCount = horizontalSnapScrollState.roomsCount,
+            columnWidth = horizontalSnapScrollState.columnWidth,
+            activeColumnIndex = horizontalSnapScrollState.activeColumnIndex
+        )
+        val newItemWidth = horizontalSnapScrollState.calculateColumnWidth(measuredWidth)
+        logging.d(
+            LOG_TAG,
+            """item width: $newItemWidth ${newItemWidth.toFloat() / resources.displayMetrics.density}dp"""
+        )
+        columnWidth = newItemWidth
+    }
+
+    override fun onScrollChanged(scrollX: Int, scrollY: Int, oldScrollX: Int, oldScrollY: Int) {
+        super.onScrollChanged(scrollX, scrollY, oldScrollX, oldScrollY)
+//		logging.d(LOG_TAG, "scrolled from " + oldScrollX + " to " + scrollX)
+        roomNames.scrollTo(scrollX, 0)
+    }
+
+    fun setChildScroller(titleScrollView: HorizontalScrollView) {
+        roomNames = titleScrollView
+    }
+
+    var columnWidth: Int
+        get() {
+            check(horizontalSnapScrollState.isRoomsCountInitialized()) { "The \"roomsCount\" field must be initialized before invoking \"getColumnWidth\"." }
```

Comment:
You can use a raw string here: `"""The "roomsCount" field must be initialized before invoking "getColumnWidth"."""`